termwiz/src/cell.rs

```diff
@@ -249,6 +249,7 @@ impl CellAttributes {
     bitfield!(wrapped, set_wrapped, 11);
     bitfield!(overline, set_overline, 12);
     bitfield!(semantic_type, set_semantic_type, SemanticType, 0b11, 13);
+    bitfield!(empty, set_empty, 15);
```

Comment:
The problem with comparing the cell contents for space is that "fake" cells are initializied with a space. And we don't want to copy these fake spaces at the end. For example, let's say you do `echo -e "abc  \ndef  "` and then select these two lines. With trimming disabled, it should copy `abc  \ndef  `, with the spaces at the end. With trimming enabled, it should copy `abc\ndef`. But since the terminal may have a greater width than `abc  `, there are fake cells at the end of the line that have a space as content. And we need a way to distinguish whether these cells are really filled with a space or just blank "fake" cells with a space as placeholder. I hope I could get across the problem.

Reply:
Let's say the terminal has a height of 2 and a width of 5 cells.
You execute
```
$ echo -e "abc  \nde"
```
Resulting text (note: 2 spaces after abc):
```
abc  
de
```
If you'd select the last two lines and trim is disabled, the resulting text should be
```
abc  \nde
```
with two spaces after abc. But if there is no way to distinguish between real spaces (here from the `echo` command) or just placeholder spaces by cells to fill up the width of the terminal, we cannot remove these "fake" spaces from the resulting text which becomes
```
abc  \nde   
```
with 2 spaces after abc and 3 spaces after de. But we do not want the 3 spaces after de, since they were not part of the real output of the `echo` command. But if we just compare for spaces to trim the end, the resulting text would be
```
abc\nde
```
with no spaces at all. But that is not what we want, since we disabled trimming.

Does that help?

Reply:
I see.  Is that case really important to handle? I can't think of a time where it has mattered for me!

Reply:
Well, if we wouldn't trim the fake cells at the end, and the terminal has a width of 200 cells, every selected line would have like 200 characters in total. I would rather prefer the "ignore fake spaces" solution. But I'm also okay with closing this PR.

> Takes up another bit for not very much gain vs. comparing the cell contents with space

That bit was unused anyway, so we don't waste any additional space

> Requires careful updates to ensure that it is correct. For example, escape sequences may overwrite the cell with non-blank content and cause this to be invalidated

To me, it seemed like a new `Cell` would be created if the content was overwritten. So that the `Cell`'s contents aren't modified directly. So it should be fine like this, I guess?

Reply:
Depending on the escape sequences used, and the APIs used internally, the attributes can be modified independently of the textual content of the cell.  That poses a number of questions that may not have clear answers.

What should the semantics be if an application clears the terminal to have a blue background color but spaces in the cells?  Are those considered blank, or should they be copied as spaces? I think a human would consider them to be logically blank, but depending on how the application painted the screen we may not have enough information to make the correct judgement.

There are a lot of situations like that that make this sort of thing more complex.

A related area to this is semantic zones: there are already a couple of hacks in the code that deal with there being "holes" in the zones due to the way that various TUI libraries optimize painting the screen and skip updating ranges of cells; what may appear to be a logical run of contiguous cells may not be painted contiguously at all.

So: I'm a bit hesitant about adding an attribute for this because there are a lot of cases that need consideration in order for it to be robust.

I'm not sure if it is worth adding complexity for this feature based on my personal experience: I don't think trailing spaces have ever been desirable when copying and pasting.

Reply:
Okay, I will close this PR then.